### ISSUE TITLE
Feedback on new PR review flow

### ISSUE THREAD

**Issue by derekprior** [arthurnn/meta#34]
Let me start by saying I'm very excited that GitHub clearly views PR reviews as a critical feature of the platform and is interested in improving it. I have a preference for the way things were before, but overall I'm excited that really smart people are excitedly working to improve GitHub, which makes my life as a developer better every day.

In my 10+ years of experience with code reviews, I realized that on some teams code reviews were something I looked forward to and others they were something I dreaded. A couple of years ago I decided to really dig into why this is, and I wrote [Cultivating a Code Review Culture](https://www.youtube.com/watch?v=PJjmw9TRB7s), a talk I've now given several times at conferences, meet-ups, and privately to companies. In my consulting work, I often work with teams to implement or improve their code reviews. All this to say, I've talked to and with a lot of people specifically about code review.
## General Feedback

The myriad problems I hear about and experience with broken code review processes have some common themes:
- A tone of approval and rejection rather than collaboration in reviews.
- Prescriptive processes that attempt to solve people problems with computers.

It's my opinion that the recent changes to pull request reviews on GitHub encourage both of these anti-patterns. 

Instead of leaving a series of comments, I'm now encouraged to "comment", "approve" or "request changes". In all my recent PR reviews, I'm having trouble thinking of a situation where I felt like I needed to stipulate that I was requesting changes. Instead, I'll ask pointed questions about code and trust my teammates to consider them and make the decision they think is best. It's this idea of approval and rejection ("request changes") that I've heard over and over again as a source of stress and tension in reviews. With the new workflow, I always choose "comment" but this leaves the author wondering why I didn't select "approve".

The changes to the PR process also layer on protections designed to keep unreviewed code from being merged. I certainly applaud this in theory as I think all changes should be reviewed, but backing this up with computers and approval encourages the emergence of gatekeepers - a process I've heard frustrations with time and again.

It's simple enough as a team to agree on what "comment" means in reviews or to agree not to use the protected branch features of the new review process for these reasons, but my concern is that their existence pushes the default GitHub review process from something very conversational (in my opinion, much to its benefit) to something very enterprise-esque. 
## UI Feedback

Please bring back collapsable outdated diffs! I know GitHub has likely heard this thousands of times already, but no feedback on the new workflow would be complete without it. I fear that perhaps this will end up with authors needing to "complete" or "reject" feedback, which I hope isn't the case for the reasons I outlined above.

I really like the ability to batch up my comments and send them as one unit. It lets me confidently add feedback I'm thinking of "in the moment" but I know might be addressed later on. I used to replicate this by leaving all of my comments un-submitted until I finished, but this is an improvement. It's also less email and notification noise, which is greatly appreciated.

The distinction between "add single comment" and "start review" is a bit puzzling. It's another decision I've got to make, along with the decision at the end of a review (if I choose that route) about whether I'm approving, requesting changes, or just commenting. Ultimately, I don't really understand what the functional difference is between the options.

**Comment by arthurnn** [comment]
hey @derekprior , thanks a lot for your feedback. I forward that to the team that is working on this, and I hope we will have something to reply here sooner rather than later.
Thanks again ❤️

**Comment by rsanheim** [comment]
❤️ ❤️ ❤️ 

Thanks for the thoughtful feedback. I'm a member of the code review team and really appreciate it.

re: prescriptive process vs GitHub's earlier, looser process - there is a lot that can be said there, and maybe it would be better suited for a discussion over a drink sometime. 😏  I think what I'll say right now is that we have to balance contexts like a Thoughtbot consultancy, massive OSS orgs, and enterprises with thousands of engineers, with everything in between. Those contexts all vary wildly, and I hope we can continue to iterate and respond to find something that works really well for all of them.

re: outdated comments collapsing - they are coming back :soon:. 👍

**Comment by derekprior** [comment]
> re: prescriptive process vs GitHub's earlier, looser process - there is a lot that can be said there, and maybe it would be better suited for a discussion over a drink sometime.

Certainly! Even in preparing this feedback this evening I thought, "There's a lot more nuance here than I have time to write up and I could do this much better in an actual conversation", but I'm thankful for the opportunity to give feedback longer than a tweet nonetheless. 

Thanks for taking the time to read.
